### PR TITLE
[CBRD-25032] Add a dump function and stat logging in debug mode

### PR DESCRIPTION
**Pull request by Rudeus**
http://jira.cubrid.org/browse/CBRD-25032

 - Implement dump functions of memory_monitor class on the server
 - Add debug log when add_stat() function adds stat.

### CODE REVIEW

File: src/base/memory_monitor_sr.cpp [hunk 1]
```diff
@@ -177,6 +182,7 @@ namespace cubperf
       void aggregate_module_info (int module_index, std::vector<MMON_MODULE_INFO> &info) const;
       void aggregate_module_info_summary (std::vector<MMON_MODULE_INFO> &info) const;
       void aggregate_tran_info (int tran_count, MMON_TRAN_INFO &info) const;
+      void stat_dump_debug () const;
```
Comment (hornetmj) [inline src/base/memory_monitor_sr.cpp:186]: wrap around to #if !defined (NDEBUG)

File: src/base/memory_monitor_sr.cpp [hunk 2]
```diff
@@ -618,6 +644,78 @@ namespace cubperf
   {
     m_aggregater.get_transaction_info (tran_count, info);
   }
+
+  void memory_monitor::stat_dump_debug () const
```
Comment (hornetmj) [inline src/base/memory_monitor_sr.cpp:663]: wrap around to #if !defined (NDEBUG)

**Review by hornetmj — APPROVED** [review]

**Review by wrlawodms — COMMENTED** [review]
How about adding a system parameter for this like `er_log_mmon_debug`, which enables to print. Without it, this is always printed. The debug msg here seems too long to be printed always.

**Comment by wrlawodms** [comment]
> How about adding a system parameter for this like er_log_mmon_debug, which enables to print. Without it, this is always printed. The debug msg here seems too long to be printed always. 

Plus, in terms of purpose, this message is not for general inquiry, but for some specific test cases, isn't it?

File: src/base/memory_monitor_sr.cpp (hunk 1 above)
Comment (Rudeus) [reply src/base/memory_monitor_sr.cpp:186]: Done.

File: src/base/memory_monitor_sr.cpp (hunk 2 above)
Comment (Rudeus) [reply src/base/memory_monitor_sr.cpp:663]: Done.

**Comment by Rudeus** [comment]
https://github.com/CUBRID/cubrid/pull/4725#pullrequestreview-1675528395

**Comment by Rudeus** [comment]
https://github.com/CUBRID/cubrid/pull/4725#pullrequestreview-1675528395
@wrlawodms 
The dump log is generated if server satisfies all two conditions: "when Memory Monitoring option is enabled" and "when the application is built in debug mode". Furthermore, it's only printed once when the server is shut down. I believe that the situation where both of these conditions are met is unlikely to occur while the server is actively used by real users. As you mentioned, the dump function is designed to be removed after all modules are registered with the Memory Monitoring Manager. Therefore, I think it's best not to create additional system parameters.

**Comment by wrlawodms** [comment]
> Furthermore, it's only printed once when the server is shut down. I believe that the situation where both of these conditions are met is unlikely to occur while the server is actively used by real users.

`[MMON-ADD]` msg is printed frequently. It would make it hard to investigate the log by developers. Do I get it wrong?

> As you mentioned, the dump function is designed to be removed after all modules are registered with the Memory Monitoring Manager

Will the functions themselves be removed? I thought that the code will be removed on #4678. If it will be removed, what's the purpose of this PR? No offense, I just don't undestarnd the purpose.

**Comment by Rudeus** [comment]
> `[MMON-ADD]` msg is printed frequently. It would make it hard to investigate the log by developers. Do I get it wrong?

You're right. That description is about dump function. Nevertheless, running the server with the Memory Monitoring option enabled and built in debug mode is a situation that can be considered to require further analysis.

> Will the functions themselves be removed? I thought that the code will be removed on #4678. If it will be removed, what's the purpose of this PR? No offense, I just don't undestarnd the purpose.

The removal of the function will be carried out through a separate additional PR. The reason this function should be retained until then is that it's necessary to verify whether the registration process for any module with the Memory Monitoring Manager is successful. Even after registering the heap module in this project, there are still pending registrations for additional modules like the lock module and transaction manager module. Therefore, it's not appropriate to question "What's the purpose of this PR?" that it will be removed in the future.

**Comment by wrlawodms** [comment]
> Even after registering the heap module in this project, there are still pending registrations for additional modules like the lock module and transaction manager module. Therefore, it's not appropriate to question "What's the purpose of this PR?" that it will be removed in the future.

OK, then it's not going to be removed in the current phase. I thought it's not worth the discussion if it will be removed soon.
Then, I am still insisting to add the system parameter not to make the debug logging cluttered. The dump at the end could be accepted if it's only printed when the mmon is enabled, but I think the frequent `[MMON-ADD]` is not acceptable.

**Review by joohok — APPROVED** [review]

**Review by wrlawodms — COMMENTED** [review]
just to change the review status.

**Comment by Rudeus** [comment]
Add system parameter and add condition in printing logs and dump

**Review by wrlawodms — APPROVED** [review]

**Pull request merged**